docs/tabular_coordinates.rst

```diff
@@ -0,0 +1,106 @@
+
+The FITS WCS standard also supports tabular axes with the ``-TAB`` CTYPE.
+Support for reading files using this convention has (reasonably) recently been added to Astropy, so if you have a FITS file using this convention you should be able to load it into a `~astropy.wcs.WCS` object.
+If you wish to be able to serialise your NDCube object to FITS files you will need to manually construct a WCS object using the ``-TAB`` convention.
+
+The functionality provided by ndcube makes it easy to construct a `gwcs.WCS` object backed by lookup tables.
+At the time of writing there are some known issues with the support for generic lookup tables in gwcs.
+
+
+Constructing a WCS from Lookup Tables
+=====================================
+
+ndcube supports constructing lookup tables from `~astropy.coordinates.SkyCoord`,  `~astropy.time.Time` and `~astropy.units.Quantity` objects.
+These objects are wrapped in `.BaseTableCoordinate` objects which can be composed together into a multi-dimensional WCS.
+
+.. note::
+
+   Only one dimensional tables are currently supported. It is possible to construct higher dimensional lookup tables by "meshing" the inputs, which is described below.
+
+A simple example of constructing a WCS from a lookup table is the following temporal axis::
+
+  >>> from astropy.time import Time
+  >>> import astropy.units as u
+  >>> from ndcube.extra_coords import TimeTableCoordinate
+
+  >>> time_axis = Time("2021-01-01T00:00:00") + (list(range(10)) * u.hour)
+  >>> time_axis
+  <Time object: scale='utc' format='isot' value=['2021-01-01T00:00:00.000' '2021-01-01T01:00:00.000'
+   '2021-01-01T02:00:00.000' '2021-01-01T03:00:00.000'
+   '2021-01-01T04:00:00.000' '2021-01-01T05:00:00.000'
+   '2021-01-01T06:00:00.000' '2021-01-01T07:00:00.000'
+   '2021-01-01T08:00:00.000' '2021-01-01T09:00:00.000']>
+
+  >>> gwcs = TimeTableCoordinate(time_axis).wcs
+  >>> gwcs
+  <WCS(output_frame=TemporalFrame, input_frame=PixelFrame, forward_transform=Model: Tabular1D
+  N_inputs: 1
+  N_outputs: 1
+  Parameters:
+    points: (<Quantity [0., 1., 2., 3., 4., 5., 6., 7., 8., 9.] pix>,)
+    lookup_table: [    0.  3600.  7200. 10800. 14400. 18000. 21600. 25200. 28800. 32400.] s
+    method: linear
+    fill_value: nan
+    bounds_error: False)>
+
+This `gwcs.WCS` object can then be passed to the constructor of `.NDCube` alongside your array and other parameters.
+
+
+Combining Two Coordinates into a Single WCS
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We can extend this example to be a space-space-time cube.
+In this example we are going to utilize the ``mesh=`` keyword argument for the first time.
+This keyword argument interprets the input to `.SkyCoordTableCoordinate` in a similar way to `numpy.meshgrid`.
+
+.. code-block::
+
+  >>> from astropy.coordinates import SkyCoord
+  >>> from ndcube.extra_coords import SkyCoordTableCoordinate
+
+  >>> icrs_table = SkyCoord(range(10)*u.deg, range(10)*u.deg)
```

Comment:
Suggest one of the axes in the `SkyCoord` have values of `range(10, 20)`*u.deg` so users can more easily see where the lat and lon values appear throughout the rest of the example.